src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -54,30 +54,57 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         let scope_def_id_sub = anon_reg_sub.def_id;
         let bregion_sub = anon_reg_sub.boundregion;
 
-        let ty_sup = or_false!(self.find_anon_type(sup, &bregion_sup));
+        let anon_ty_sup = or_false!(self.find_anon_type(sup, &bregion_sup));
 
-        let ty_sub = or_false!(self.find_anon_type(sub, &bregion_sub));
+        let anon_ty_sub = or_false!(self.find_anon_type(sub, &bregion_sub));
+        let (ty_sup, ty_fndecl_sup) = anon_ty_sup;
+        let (ty_sub, ty_fndecl_sub) = anon_ty_sub;
 
-        let (main_label, label1, label2) = if let (Some(sup_arg), Some(sub_arg)) =
+        let (spanfin1, spanfin2, main_label, label) = if let (Some(sup_arg),
+                                                                       Some(sub_arg)) =
```

Comment:
Nit: can we use use the flatter style here too? I really find the `if let` nesting hard to read. In other words, change the `if let (Some(sup_arg), Some(sub_arg)) = (...) { ... } else { false }` too:

```
let sup_arg = or_false!(self.find_arg_with_anonymous_region(sup, sup));
let sub_arg = or_false!(self.find_arg_with_anonymous_region(sub, sub));
...
```